tool/movement.re

```diff
@@ -0,0 +1,138 @@
+static mut FLYING_UP_DOWN_VELOCITY_LABEL = Text { text: "Up/Down Flying Velocity" };
+static mut FLYING_FORWARD_BACKWARD_VELOCITY_LABEL = Text { text: "Forward/Backward Flying Velocity" };
+
+static mut MOVEMENT_STATE = MovementState {
+    is_flying_down: false,
+    is_flying_up: false,
+    enable_fly: false,
+    fly_down_up_velocity: 300.,
+};
+
+struct MovementState {
+    is_flying_down: bool,
+    is_flying_up: bool,
+    enable_fly: bool,
+    fly_down_up_velocity: float,
+}
+
+fn create_movement_menu() -> Ui {
+    let mut movement_mode = Number { number: 1 };
+    Ui::new("Movement:", List::of(
+        UiElement::Button(UiButton {
+            label: Text { text: "Enable Flying" },
+            onclick: fn(label: Text) {
+                MOVEMENT_STATE.enable_fly = true;
+                leave_ui();
+                leave_ui();
+                leave_ui();
+            },
+        }),
+        UiElement::Button(UiButton {
+            label: Text { text: "Disable Flying" },
+            onclick: fn(label: Text) {
+                MOVEMENT_STATE.enable_fly = false;
+                Tas::set_movement_mode(1);
+            },
+        }),
+        UiElement::Chooser(Chooser {
+            label: Text { text: "Movement Mode" },
+            options: List::of(
+                Text { text: "None" },
+                Text { text: "Walking" },
+                Text { text: "Navwalking" },
+                Text { text: "Falling" },
+                Text { text: "Swimming" },
+                Text { text: "Flying" },
+            ),
+            selected: movement_mode.number,
+            onchange: fn(index: int) { movement_mode.number = index; },
+        }),
+        UiElement::Button(UiButton {
+            label: Text { text: "Set Movement Mode" },
+            onclick: fn(label: Text) { Tas::set_movement_mode(movement_mode.number); },
+        }),
+        UiElement::Input(Input {
+            label: FLYING_UP_DOWN_VELOCITY_LABEL,
+            input: "300",
+            onclick: fn(input: string) {
+                let val = input.parse_float();
+                match val {
+                    Result::Ok(val) => { MOVEMENT_STATE.fly_down_up_velocity = val; },
+                    Result::Err(e) => { FLYING_UP_DOWN_VELOCITY_LABEL.text = "Up/Down Flying Velocity [error: invalid input]"; },
+                }
+            },
+            onchange: fn(input: string) {},
+        }),
+        UiElement::Input(Input {
+            label: FLYING_FORWARD_BACKWARD_VELOCITY_LABEL,
+            input: "600",
+            onclick: fn(input: string) {
+                let val = input.parse_float();
+                match val {
```

Comment:
You could directly `match input.parse_float()`, removing a redundant assignment.